GltfExport.cs

```diff
@@ -64,93 +68,103 @@ protected override void SolveInstance(IGH_DataAccess DA)
-                        else res = "Something went wrong";
+                            RhinoApp.Write("Object(s) baked\n", true);
 
-                        break;
-                    default:
+                            // select object(s) into list
+                            foreach (Guid id in ids)
+                                doc.Objects.Select(id, true);
 
-                        // attempt export entire document scene
-                        if (FileGltf.Write(filePath, doc, optionsGltf))
-                            res = $"File written successfully to path: {filePath}";
-                        else res = "Something went wrong";
-                        break;
+                            RhinoApp.Write("Object(s) selected\n", true);
+
+                            // attempt export temp baked object(s)
+                            if (doc.ExportSelected(filePath, archiveCollection))
+                                res = $"File written successfully to path: {filePath}";
+                            else
+                                throw new Exception("Something went wrong when exporting");
+                            break;
+                        default:
+                            // attempt export entire document scene
+                            if (FileGltf.Write(filePath, doc, optionsGltf))
+                                res = $"File written successfully to path: {filePath}";
+                            else
+                                throw new Exception("Something went wrong when exporting");
+                            break;
+                    }
                 }
-            }
-            catch (Exception e)
-            {
-                res += "\nError\n" + e;
-            }
-            finally
-            {
-                RhinoApp.Write("Export attempted\n", true);
-                if (!scene)
+                catch (Exception e)
                 {
-                    // delete baked geometry quiet
-                    foreach (var id in ids)
-                        doc.Objects.Delete(id, true);
-                    RhinoApp.Write("Object(s) deleted\n", true);
+                    res += "\nError\n" + e;
                 }
-                _lastResult = res;
-            }
-            #region Outputs
+                finally
+                {
+
+                    RhinoApp.Write($"Export attempted to directory: {loc}\n", true);
+                    if (!scene)
+                    {
+                        // delete baked geometry quiet
+                        foreach (Guid id in ids)
+                            doc.Objects.Delete(id, true);
+                        RhinoApp.Write("Object(s) deleted\n", true);
+                    }
+
+                    _lastResult = $"[{DateTime.Now.ToShortTimeString()}] {res}";
+                    _executing = false;
+                    RhinoApp.InvokeOnUiThread(()=>this.ExpireSolution(true));
```

Comment:
# GH-Specific-Flow - Forcing the component to recompute
 - while I get this now, we should really add comments this (e.g., // must be called outside component thread)
 - even better if we write a class with the methods (maybe static) to deal with this
   - not to obstruct the rest of the logic 
 - [see also](#threading-gh-specific-flow-managing-concurrent-executions)